modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -292,7 +284,7 @@ public void updateMessageID(String queueName, String nodeId, long startMessageId
             firstMessageId = startMessageIdInTheSlot;
         }
 
-	    if (slotRecoveryScheduled.compareAndSet(true, true)) {
```

Comment:
I think it is. Let me explain,

a. If we use compareAndSet following are the micro steps
 1. compareAndSet will atomically compare and return boolean
 2. if condition will act according to that

b. if we use get,
 1. get will return the boolean value
 2. if condition will act according to that

Isn't it the same?